### PR TITLE
Allow only Sharezone Plus users to change the homework reminder time

### LINKED ISSUE #886: Restrict changing the homework reminder time for non Sharezone Plus users
Design: https://www.figma.com/file/EIlLPe7KdF4bQXLQVAsDeM/SharezoneApp?type=design&node-id=2380-5755&mode=design&t=XAyp3Z0WOjT8vZ58-4

### PR DESCRIPTION
**Pull request by nilsreichardt**
## Demo

image

image

## Related Tickets

Closes #886

### CODE REVIEW

File: app/lib/pages/settings/notification.dart [hunk 1]
```diff
@@ -205,6 +231,65 @@ class _HomeworkNotificationsTimeTile extends StatelessWidget {
...
+
+class _ChangeHomeworkNotificationsTimePlusAd extends StatelessWidget {
+  const _ChangeHomeworkNotificationsTimePlusAd();
+
+  @override
+  Widget build(BuildContext context) {
+    final buttonShape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.circular(12.5),
+    );
+    return AlertDialog(
```
Comment (nilsreichardt) [inline app/lib/pages/settings/notification.dart:254]: I'll later make a more abstract widget in the `sharezone_widgets` package to be able to reuse this dialog for other features. Later on, we can also implement in this dialog the logic for displaying a short demo video (mobile video for mobile devices and landscape video for landscape devices).

Comment (nilsreichardt) [reply app/lib/pages/settings/notification.dart:254]: https://github.com/SharezoneApp/sharezone-app/pull/1008

File: app/lib/pages/settings/notification.dart [hunk 2]
```diff
@@ -169,6 +173,17 @@ class _HomeworkNotificationsTimeTile extends StatelessWidget {
     });
   }
 
+  Future<void> showPlusAdDialog(BuildContext context) async {
+    final shouldShowPlusAd = await showDialog<bool>(
```
Comment (Jonas-Sander) [inline app/lib/pages/settings/notification.dart:177]:
```suggestion
    final shouldNavigateToPlusPage = await showDialog<bool>(
```

File: app/test/pages/settings/notification_page_test.dart [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+              home: NotificationPage(),
+            ),
+          ),
+        ),
+      );
+    }
+
+    group('homeworks', () {
+      testWidgets(
+          'does not display homework notification section non student accounts',
```
Comment (Jonas-Sander) [inline app/test/pages/settings/notification_page_test.dart:71]:
```suggestion
          'does not display the homework notification section for non-student accounts',
```

File: app/test/pages/settings/notification_page_test.dart [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+          typeOfUser: TypeOfUser.student,
+        );
+
+        expect(
+          find.byKey(const ValueKey('homework-notifications-card')),
+          findsOneWidget,
+        );
+      });
+
+      testWidgets('display Sharezone Plus chip for non plus accounts',
```
Comment (Jonas-Sander) [inline app/test/pages/settings/notification_page_test.dart:100]:
```suggestion
      testWidgets('displays Sharezone Plus chip for non-plus accounts',
```

File: app/test/pages/settings/notification_page_test.dart [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        );
+
+        expect(
+          find.byType(SharezonePlusChip),
+          findsNothing,
+        );
+      });
+
+      testWidgets(
+          'show sharezone plus ad when tapping on homework reminder time tile',
```
Comment (Jonas-Sander) [inline app/test/pages/settings/notification_page_test.dart:135]:
```suggestion
          'shows sharezone plus ad when tapping on homework reminder time tile',
```

**Review by Jonas-Sander — APPROVED** [review]

**Comment by Jonas-Sander** [comment]
Looks nice 👍

**Pull request merged**